akka-http-core/src/test/scala/akka/http/impl/engine/server/HttpServerSpec.scala

```diff
@@ -1009,7 +906,7 @@ class HttpServerSpec extends AkkaSpec(
         netIn.sendComplete()
         netOut.cancel()
 
-        requests.expectComplete()
+        requests.expectError()
```

Comment:
What happens here is that these scenarios depended on cancellation eventually completing the controller stage which would propagate to the user handler. The new cancellation strategy is "fail after delay". So now the user handler is failed in these situations.

This is somewhat questionable. In most cases, the user handler flow does not really care about stream completion / failures and will just propagate it, but if it does it will now see a somewhat opaque NoMoreElementsNeeded failure. This could be fixed by using the 2.6-only cancellation strategy which will convert those cancellations into completions. Or, we need to see how to change the server-side not to depend on cancellation in these scenarios.

Reply:
Hmmmm, changing this to change it again when we use "the 2.6-only cancellation strategy" sounds _not great_. Should we hold this PR until we're 2.6+ only?

Reply:
No, then we'd keep the bugs until another year when we drop 2.6 support.